src/pages/ReplaceServer/InstanceMigrationMonitor.js

```diff
@@ -0,0 +1,192 @@
+* See the License for the specific language governing permissions and
+* limitations under the License.
+**/
+
+import React, { Component } from 'react';
+import { ProgressBar } from 'react-bootstrap';
+import { translate } from '../../localization/localize.js';
+import { ActionButton } from '../../components/Buttons.js';
+import { fetchJson } from '../../utils/RestUtils.js';
+import { ErrorMessage } from '../../components/Messages.js';
+import { ConfirmModal } from '../../components/Modals.js';
+
+// this the modal dialog for user to monitor the instances live
+// migration results
+class InstanceMigrationMonitor extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      loading: false,
+      errorMsg: undefined,
+      migrated: 0,
+      migrating: props.data.length,
+      now: new Date()
+    };
+    // time out after 30 minutes
+    this.expired = new Date(this.state.now);
+    this.expired.setMinutes(this.state.now.getMinutes() + 30);
+  }
+
+  componentDidMount() {
+    this.fetchInstances();
+    this.timer = setInterval(() => this.fetchInstances(), 5000);
+  }
+
+  componentWillUnmount() {
+    // unset polling timer
+    if(this.timer) {
+      clearInterval(this.timer);
+      this.timer = undefined;
+    }
+  }
+
+  fetchInstances = () => {
+    if(this.isTimeOut() || this.state.migrated === this.props.data.length) {
+      // unset polling timer
+      if(this.timer) {
+        clearInterval(this.timer);
+        this.timer = undefined;
+      }
+      return;
+    }
+
+    this.setState({now: new Date()});
+    if (!this.state.loading) {
+      let apiUrl =
+        '/api/v1/clm/compute/instances/' + this.props.operationProps.oldServer.hostname;
+      this.setState({loading: true});
+      fetchJson(apiUrl)
+        .then(response => {
+          this.setState({
+            migrating: response.length,
+            migrated: this.props.data.length - response.length,
+            loading: false
+          });
+        })
+        .catch((error) => {
+          let msg = translate('server.deploy.progress.get_instances.error', error.toString());
+          this.setState({loading: false, errorMsg: msg});
+          if(this.timer) {
+            clearInterval(this.timer);
+            this.timer = undefined;
+          }
+        });
+    }
+  }
+
+  handleDone = () => {
+    this.props.doneAction({
+      migrated: this.state.migrated, migrating: this.state.migrating,
+      timeout: this.isTimeOut()});
+  }
+
+  handleCancel = () => {
+    this.props.cancelAction({
+      migrated: this.state.migrated, migrating: this.state.migrating,
+      timeout: this.isTimeOut()});
+  }
+
+  handleHide = () => {
+    if(this.state.migrated === this.props.data.length) {
+      this.handleDone();
+    }
+    else {
+      this.handleCancel();
+    }
+  }
+
+  isTimeOut = () => {
+    return this.state.now.getTime() > this.expired.getTime();
```

Comment:
this usage makes the name "now" even more confusing, if we really wanted to compare to "now" wouldnt we just crank out a new Date object and pull the time off of that? is "now" only present to avoid instantiating too many Dates? I'm confused

Reply:
@jeremy-moffitt in the fetchInstances method, if it can fetch instances, I reset now...and isTimeOut is used to disable done button...if it is timeout and all the instances are not fully migrated.